src/DurableTask.SqlServer.AzureFunctions/SqlDurabilityProvider.cs

```diff
@@ -200,13 +197,12 @@ public override bool TryGetScaleMonitor(
             string storageConnectionString,
             out IScaleMonitor scaleMonitor)
         {
-            if (this.scaleMonitor == null)
+            if (this.singletonSqlMetricsProvider == null)
             {
-                var sqlMetricsProvider = new SqlMetricsProvider(this.service);
-                this.scaleMonitor = new SqlScaleMonitor(hubName, sqlMetricsProvider);
+                this.singletonSqlMetricsProvider = new SqlMetricsProvider(this.service);
             }
-
-            scaleMonitor = this.scaleMonitor;
+            scaleMonitor = new SqlScaleMonitor(hubName, this.singletonSqlMetricsProvider);
```

Comment:
Based on the changes Alexey made in the Azure Storage repo, I think we also need to include the `functionId` value in the monitor that we create. That way each monitor has a unique name/ID.